src/lib/tradingView/utils.ts

```diff
@@ -8,21 +8,37 @@ import { Themes } from '@/styles/themes';
 
 import { AppTheme, type AppColorMode } from '@/state/configs';
 
+import { testFlags } from '../testFlags';
+
 export const mapCandle = ({
   startedAt,
   open,
   close,
   high,
   low,
   baseTokenVolume,
-}: Candle): TradingViewBar => ({
-  time: new Date(startedAt).getTime(),
-  low: parseFloat(low),
-  high: parseFloat(high),
-  open: parseFloat(open),
-  close: parseFloat(close),
-  volume: Math.ceil(Number(baseTokenVolume)),
-});
+  trades,
+  orderbookMidPriceOpen,
+  orderbookMidPriceClose,
+}: Candle): TradingViewBar => {
+  const isOhlcEnabled = testFlags.ohlc;
+  const hasNoTrades = trades === 0;
+
+  return {
+    time: new Date(startedAt).getTime(),
+    low: parseFloat(low),
+    high: parseFloat(high),
+    open:
```

Comment:
does this mean we're switching back and forth between data sources per-candle? Don't we kind of want all-or-nothing?

Reply:
from my understanding it is actually per candle (at least until we implement the toggle), but will double check!